compiler/rustc_typeck/src/check/op.rs

```diff
@@ -549,16 +549,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         is_assign: IsAssign,
         op: hir::BinOp,
     ) -> bool {
-        let source_map = self.tcx.sess.source_map();
-        let remove_borrow_msg = "String concatenation appends the string on the right to the \
-                                 string on the left and may require reallocation. This \
-                                 requires ownership of the string on the left";
-
-        let msg = "`to_owned()` can be used to create an owned `String` \
-                   from a string reference. String concatenation \
-                   appends the string on the right to the string \
-                   on the left and may require reallocation. This \
-                   requires ownership of the string on the left";
+        let str_concat_note = "String concatenation appends the string on the right to the
+string on the left and may require reallocation.
+This requires ownership of the string on the left.";
+        let rm_borrow_msg = "remove the borrow to obtain an owned `String`";
+        let to_owned_msg = "use `to_owned()` to create an owned `String` from a string reference";
```

Comment:
```suggestion
        let to_owned_msg = "create an owned `String` from the borrowed string";
```

I don't know that we necessarily need to reference `to_owned` here at all, that's in the suggestion itself, we can just describe what it is doing.

Reply:
Sounds good to me.